src/machines/modelingMachine.ts

```diff
@@ -663,7 +680,10 @@ export const modelingMachine = setup({
     },
     'assign tool in context': assign({
       currentTool: ({ event }) =>
-        'data' in event && event.data && 'tool' in event.data
+        'data' in event &&
+        event.data &&
+        'tool' in event.data &&
+        !('target' in event.data)
```

Comment:
Change tool got effected because `'tool' in event.data` is not sufficient type narrowing since tool is also used in subtract.